### ISSUE TITLE
Allow different `text_embedder_cfg` for each column

### ISSUE THREAD

**Issue by toenshoff** [pyg-team/pytorch-frame#206]
Currently, the `Dataset` class expects a single `text_embedder_cfg` object that is used globally to embed all text embedding columns.

This may not be desirable in cases where different pre-trained models should be applied to different columns.
An important example would be multi-lingual data where different text columns have different languages, say English and Chinese.

I think it would be great if one can also optionally pass a dictionary with columns-wise configs, similar to the `col_to_sep` argument.
I would gladly add this feature if it is considered helpful.

Thanks for this project by the way, it will be very useful for many datasets.

**Comment by toenshoff** [comment]
I see, that makes sense. 

Would it be reasonable to add the embedding dimension as an attribute to `StatType`? 
It may be cumbersome to manage this information separately from other column statistics,  especially when there are multiple embedded columns with varying dimensions.

**Comment by weihua916** [comment]
Yes, that makes sense. We were working towards that direction, but the challenge was that column stats are computed purely based on the pandas columns (`ser` [here](https://github.com/pyg-team/pytorch-frame/blob/master/torch_frame/data/dataset.py#L430-L431)). We cannot obtain the embedding dimensionality information until we actually apply the embedder, which is why we let our users manually specify the dimensionality.

Do you have any other ideas?

**Comment by toenshoff** [comment]
I agree that this is a little tricky. 

The simplest solution may be to add a `dim` attribute to `TextEmbedderConfig` which enables the user to specify the embedding dimension in the `text_embedder_cfg`  objects that are passed. This could then be passed on as a new argument to `compute_col_stats` [here](https://github.com/pyg-team/pytorch-frame/blob/e8a11d43746d6b69b97e1d3646451d8a88a9ab88/torch_frame/data/dataset.py#L431C1-L431C64) just like the optional separators already are. As a downside, this would clutter the signature of `compute_col_stats` and a user may provide the wrong embedding dimension.

A second solution would be to add a new method that updates existing `col_stats` given a `TensorFrame` object. So something like:
```python
update_col_stats(stats: Dict[StatType, Any], tensor_frame: TensorFrame) -> Dict[StatType, Any]
```
This could be called for the stats of each column after materializing the frame to add further information such as embedding dimensions.

Let me know if these suggestions make any sense.

**Comment by weihua916** [comment]
These are great suggestions. I think the second solution is quite promising, especially because we no longer requires our users to specify any dimensionality information! We can insert this right after `TensorFrame` creation [here](https://github.com/pyg-team/pytorch-frame/blob/master/torch_frame/data/dataset.py#L442-L444).

**Comment by weihua916** [comment]
@toenshoff I adopted your idea of `update_col_stats` in https://github.com/pyg-team/pytorch-frame/pull/222! thanks. 

We will support your request once `MultiEmbeddedTensor` is ready.

**Comment by toenshoff** [comment]
Looks good, thank you!
